cmake/kokkos_settings.cmake

```diff
@@ -158,6 +158,19 @@ if (NOT "${KOKKOS_INTERNAL_ADDTOPATH}" STREQUAL "")
   set(KOKKOS_SETTINGS ${KOKKOS_SETTINGS} "PATH=\"${KOKKOS_INTERNAL_ADDTOPATH}:$ENV{PATH}\"")
 endif()
 
+if (CMAKE_CXX_STANDARD)
+  if (CMAKE_CXX_STANDARD STREQUAL "98")
+    message(FATAL_ERROR "Kokkos requires C++11 or newer!")
+  endif()
+  set(KOKKOS_CXX_STANDARD "c++${CMAKE_CXX_STANDARD}")
+  if (CMAKE_CXX_EXTENSIONS)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+      set(KOKKOS_CXX_STANDARD "gnu++${CMAKE_CXX_STANDARD}")
```

Comment:
I don't think we want this. We want proper C++ standard not GNU extensiosn. In fact I think we don't really support that.

Reply:
I can remove that, but then downstream users of Kokkos who have `CMAKE_CXX_EXTENSIONS=ON` will get both `-std=gnu++11 -std=c++11` in their flags.